File: app/src/main/java/com/typ/muslim/interfaces/KhatmaManagerCallback.java
/*
 * This app is developed by AHMED SLEEM
 *
 * Copyright (c) 2021.  TYP INC. All Rights Reserved
 */

package com.typ.muslim.interfaces;

import androidx.annotation.NonNull;

import com.typ.muslim.models.Khatma;

public interface KhatmaManagerCallback {


    /**
     * Called when KhatmaManager instance is created and is in constructing and initializing state
     */
    void onPrepareManager();

    /**
     * Called when KhatmaManager puts an Khatma under its management
     *
     * @param underManagementKhatma The under manager management khatma
     */
    void onPutUnderManagement(@NonNull Khatma underManagementKhatma);

    /**
     * Called when KhatmaManager hasn't found any active khatma to put under management
     */
    void onNoActiveKhatmaFound();

    /**
     * Called when the currently under-management active khatma has made a new progress on it.
     */
    void onKhatmaProgressUpdated();

    /**
     * Called when KhatmaManager has released under-management khatma from being managed anymore
     */
    void onKhatmaReleased();

}

File: app/src/main/java/com/typ/muslim/features/khatma/interfaces/KhatmaManagerCallback.kt
/*
 * This app is developed by AHMED SLEEM
 *
 * Copyright (c) 2021.  TYP INC. All Rights Reserved
 */
package com.typ.muslim.features.khatma.interfaces

import com.typ.muslim.features.khatma.models.Khatma
import com.typ.muslim.features.khatma.models.KhatmaWerd

interface KhatmaManagerCallback {

    /**
     * Called when KhatmaManager instance is created initializing its constructor.
     */
    fun onPrepareManager()

    /**
     * Called when KhatmaManager puts an Khatma under its management
     *
     * @param khatma The under manager management khatma
     */
    fun onManageKhatma(khatma: Khatma)

    /**
     * Called when KhatmaManager deletes a khatma or
     */
    fun onNoActiveKhatma()

    /**
     * Called when the currently under-management active khatma has made a new progress on it.
     */
    fun onProgressUpdated(nextWerd: KhatmaWerd)

    /**
     * Called when KhatmaManager switches from this khatma to another one
     */
    fun onSwitchKhatma(khatma: Khatma)

    fun onFinishKhatma()

}